### PR TITLE
Add Client.CountWorkflowsAsync

### LINKED ISSUE #30: Add API to count workflows
See parent issue for more information.

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

Added `CountWorkflowsAsync` call for counting workflows

## Checklist

1. Closes #30

### CODE REVIEW

File: src/Temporalio/Client/WorkflowExecutionCount.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+            /// <summary>
+            /// Gets the approximate number of workflows matching the original query for this group.
+            /// </summary>
+            public long Count { get; private init; }
+
+            /// <summary>
+            /// Gets the search attribute values for this group.
+            /// </summary>
+            public IReadOnlyCollection<object> GroupValues { get; private init; }
```
Comment (cretz) [inline src/Temporalio/Client/WorkflowExecutionCount.cs:63]: I have an outstanding question to the server team on why this is a collection in the API. Will not merge until I understand it.

File: src/Temporalio/Client/WorkflowCountOptions.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Options for <see cref="ITemporalClient.CountWorkflowsAsync" />.
+    /// </summary>
+    public class WorkflowCountOptions : ICloneable
+    {
+        /// <summary>
+        /// Gets or sets RPC options for starting the workflow.
```
Comment (Sushisource) [inline src/Temporalio/Client/WorkflowCountOptions.cs]: Copy paste mistake

**Review by Sushisource — APPROVED** [review]

File: src/Temporalio/Client/WorkflowExecutionCount.cs (hunk 1 above)
Comment (cretz) [reply src/Temporalio/Client/WorkflowExecutionCount.cs:63]: Confirmed, it makes sense for when there are multiple group-by values

File: src/Temporalio/Client/WorkflowCountOptions.cs (hunk 2 above)
Comment (cretz) [reply src/Temporalio/Client/WorkflowCountOptions.cs]: Fixed

**Pull request merged**